types.cc

```diff
@@ -843,7 +842,7 @@ abstract_type::cql3_kind abstract_type::get_cql3_kind_impl() const {
         cql3_kind operator()(const simple_date_type_impl&) { return cql3_kind::DATE; }
         cql3_kind operator()(const utf8_type_impl&) { return cql3_kind::TEXT; }
         cql3_kind operator()(const time_type_impl&) { return cql3_kind::TIME; }
-        cql3_kind operator()(const timestamp_type_impl&) { return cql3_kind::TIMESTAMP; }
+        cql3_kind operator()(const concrete_type<db_clock::time_point>&) { return cql3_kind::TIMESTAMP; }
```

Comment:
Just above, you replaced `date_type_impl` with `concrete_type<db_clock::time_point>`, now you do the same with `timestamp_type_impl`. Are the two types essentially the same now? I'm confused.

Reply:
It is the common base class. The one difference between date_type and timestamp_type is how they are compared.